src/Groups/pcgroup.jl

```diff
@@ -923,3 +923,57 @@ function collector(::Type{T}, G::PcGroup) where T <: IntegerUnion
 end
 
 collector(G::PcGroup) = collector(ZZRingElem, G)
+
+# GAP wrappers for group encoding / decoding
+
+"""
+    code_pcgroup(G::PcGroup)
+
+Return a `ZZRingElem` representing the polycyclic group `G`, using the same
+encoding as GAP's `CodePcGroup` and Magma's `SmallGroupEncoding`.
+Currently only defined for `PcGroup`, not `SubPcGroup`.
+
+# Examples
+```jldoctest
+julia> G = small_group(12, 2)
+Pc group of order 12
+
+julia> code = code_pcgroup(G)
+266
+
+julia> H = pcgroup_code(code, order(G))
+Pc group of order 12
+
+julia> code_pcgroup(G) == code_pcgroup(H)
+true
+```
+"""
+function code_pcgroup(G::PcGroup)
+  return ZZ(GAP.Globals.CodePcGroup(GapObj(G))::GapInt)
+end
+
+"""
+   pcgroup_code(code::IntegerUnion, order::IntegerUnion)
+
+Given an integer `code` and an integer `order`, return the polycyclic group it encodes.
+Both `code` and `order` can be of type `Int`, `BigInt`, or `ZZRingElem`. Internally, these are converted to `GapInt` before calling the GAP function `PcGroupCode`.
+The accepted codes and resulting groups match those of GAP's `PcGroupCode` and Magma's `SmallGroupDecoding`.
+
+# Examples
+```jldoctest
+julia> G = small_group(12, 2)
+Pc group of order 12
+
+julia> code = code_pcgroup(G)
+266
+
+julia> H = pcgroup_code(code, order(G))
+Pc group of order 12
+
+julia> code_pcgroup(G) == code_pcgroup(H)
+true
+```
+"""
+function pcgroup_code(code::IntegerUnion, order::IntegerUnion)
+  return PcGroup(GAP.Globals.PcGroupCode(GAP.GapInt(BigInt(code)),GAP.GapInt(BigInt(order))))
```

Comment:
As @lgoettgens pointed out this is inefficient. We will add a `GAP.GapInt` constructor in GAP.jl so that one can directly write `GAP.GapInt(code)`

Reply:
See https://github.com/oscar-system/GAP.jl/pull/1283. Once that and https://github.com/oscar-system/GAP.jl/pull/1284 are merged (and further 20 minutes have passed), this line can be changed to
 ```suggestion
   return PcGroup(GAP.Globals.PcGroupCode(GapInt(code), GapInt(order)))
```
At the same time, please change https://github.com/oscar-system/Oscar.jl/blob/c0fadd1a7ff95ba76841972ba5ca5122f5a5aba0/Project.toml#L38 to "0.16.1"

Reply:
GAP.jl 0.16.1 is out now, so this can be done.